libraries/AC_PID/AC_PID.cpp

```diff
@@ -64,12 +66,62 @@ const AP_Param::GroupInfo AC_PID::var_info[] = {
     // @User: Advanced
     AP_GROUPINFO_FLAGS_DEFAULT_POINTER("SMAX", 12, AC_PID, _slew_rate_max, default_slew_rate_max),
 
+#if AC_PID_ADVANCED_ENABLED
+    // @Param: ADV
+    // @DisplayName: Advanced parameters enable
+    // @Description: Advanced parameters enable
+    // @Values: 0:Disabled,1:Enabled
+    // @User: Advanced
+    AP_GROUPINFO_FLAGS("ADV", 32, AC_PID, _adv_enable, 0, AP_PARAM_FLAG_ENABLE),
+
+    // @Param: D_FF
+    // @DisplayName: PID Derivative FeedForward Gain
+    // @Description: FF D Gain which produces an output that is proportional to the rate of change of the target
+    // @Range: 0 0.02
+    // @Increment: 0.0001
+    // @User: Advanced
+    AP_GROUPINFO_FLAGS_DEFAULT_POINTER("D_FF", 33, AC_PID, _kdff, default_kdff),
+
+    // @Param: NTF
+    // @DisplayName: PID Target notch Filter center frequency
+    // @Description: PID Target notch Filter center frequency in Hz.
+    // @Range: 10 495
+    // @Units: Hz
+    // @User: Advanced
+    AP_GROUPINFO("NTF", 34, AC_PID, _notch_T_center_freq_hz, 0),
+
+    // @Param: NEF
+    // @DisplayName: PID Error notch Filter center frequency
+    // @Description: PID Error notch Filter center frequency in Hz.
+    // @Range: 10 495
+    // @Units: Hz
+    // @User: Advanced
+    AP_GROUPINFO("NEF", 35, AC_PID, _notch_E_center_freq_hz, 0),
+
+    // @Param: NBW
+    // @DisplayName: PID notch Filter bandwidth
+    // @Description: PID notch Filter bandwidth in Hz.
+    // @Range: 5 250
+    // @Units: Hz
+    // @User: Advanced
+    AP_GROUPINFO("NBW", 36, AC_PID, _notch_bandwidth_hz, 0),
+
+    // @Param: NATT
+    // @DisplayName: PID notch Filter attenuation
+    // @Description: PID notch Filter attenuation in dB.
+    // @Range: 5 50
+    // @Units: dB
+    // @User: Advanced
+    AP_GROUPINFO("NATT", 37, AC_PID, _notch_attenuation_dB, 40),
```

Comment:
I will just note that there are two center frequencies here but only one set of bandwidth and attenuation. This means you have to have them both the same, or just use target or error but not both.

Reply:
Yeah, noted. I originally had all params. Interested in @lthall and @bnsgeyer 's thoughts on this

Reply:
@andyp1per and @IamPete1   
Leonard and I decided to make users use the same parameters for target and error because it was unlikely that someone would need to have the notch for the target and error signal be different. And it was more beneficial to reduce the parameter count